src/core/shared_memory_manager.cc

```diff
@@ -98,12 +105,39 @@ UnmapSharedMemory(void* mapped_addr, size_t byte_size)
   return Status::Success;
 }
 
+#ifdef TRTIS_ENABLE_GPU
+void
+OpenCUDAIPCRegion(ipcCUDA_t* shm_cuda_rep, void** data_ptr)
+{
+  cudaEvent_t event;
+
+  cudaError_t err = cudaSetDevice(shm_cuda_rep->device);
+
+  // get cuda event and synchronize
+  err = cudaIpcOpenEventHandle(&event, shm_cuda_rep->eventHandle);
+  err = cudaEventSynchronize(event);
+
+  // allocate data on the gpu and read IPC data into it
+  err = cudaMalloc(data_ptr, shm_cuda_rep->byte_size);
+  if (err != cudaSuccess) {
+    LOG_ERROR << "failed to allocate GPU memory with byte size"
+              << shm_cuda_rep->byte_size << ": "
+              << std::string(cudaGetErrorString(err));
+  }
+  err = cudaIpcGetMemHandle(
+      (cudaIpcMemHandle_t*)&shm_cuda_rep->memHandle, *data_ptr);
+
+  // set device to default GPU
+  err = cudaSetDevice(DEFAULT_GPU_ID);
```

Comment:
Not necessary for this thread. We could get the device at the beginning of the function and restore the device back to that setting, but I don't think that is necessary. The threads handling requests should not assume any particular CUDA device.

Reply:
Understood

Reply:
Done